### PR TITLE
Change nvcuvid link to dyn + hint for FFmpeg

### PR DESCRIPTION
**Pull request by Kh4L**
- User building DALI from source can now provide `FFMPEG_ROOT_DIR` to CMake to specify the right version to use, for systems where a wrong FFmpeg version is chosen by the linker
- Change to dynamic linking for `nvcuvid`

### CODE REVIEW

**Comment by Kh4L** [comment]
Build 594317

**Comment by Kh4L** [comment]
Build 594322

**Comment by Kh4L** [comment]
Build 594329

File: dali/pipeline/operators/reader/loader/video_loader.h [hunk 1]
```diff
@@ -114,6 +114,7 @@ class VideoLoader : public Loader<GPUBackend, SequenceWrapper> {
   }
 
   void init() {
+    cuvidInit(0);
```
Comment (JanuszL) [inline dali/pipeline/operators/reader/loader/video_loader.h]: Please add some error checking here and meaningful message about `that your env required video capability. If you are using docker -  refer to https://github.com/NVIDIA/nvidia-docker/wiki/Usage`, or something similar.

Comment (Kh4L) [reply dali/pipeline/operators/reader/loader/video_loader.h]: Done

**Comment by Kh4L** [comment]
Build 594527

File: cmake/Dependencies.cmake [hunk 2]
```diff
@@ -154,12 +154,25 @@ list(APPEND DALI_EXCLUDES libprotobuf.a)
...
+      list(APPEND FFmpeg_LIBS ${m})
+    else()
+      find_library(FFmpeg_Lib ${m}
+            PATHS ${FFMPEG_ROOT_DIR}
+            PATH_SUFFIXES lib lib64
+            NO_DEFAULT_PATH)
+      list(APPEND FFmpeg_LIBS ${FFmpeg_Lib})
+      message(STATUS ${m})
+    endif()
     #list(APPEND FFmpeg_LIBS PkgConfig::${m})
```
Comment (JanuszL) [inline cmake/Dependencies.cmake]: remove

File: cmake/Dependencies.cmake [hunk 3]
```diff
@@ -171,9 +184,19 @@ list(APPEND DALI_LIBS ${avformat_LIBRARIES})
 # CHECK_STRUCT_HAS_MEMBER("struct AVStream" codecpar libavformat/avformat.h HAVE_AVSTREAM_CODECPAR LANGUAGE C)
 set(CMAKE_EXTRA_INCLUDE_FILES libavcodec/avcodec.h)
 #check_type_size("AVBSFContext" AVBSFCONTEXT LANGUAGE CXX)
```
Comment (JanuszL) [inline cmake/Dependencies.cmake]: remove

File: cmake/Dependencies.cmake [hunk 4]
```diff
@@ -171,9 +184,19 @@ list(APPEND DALI_LIBS ${avformat_LIBRARIES})
 # CHECK_STRUCT_HAS_MEMBER("struct AVStream" codecpar libavformat/avformat.h HAVE_AVSTREAM_CODECPAR LANGUAGE C)
```
Comment (JanuszL) [inline cmake/Dependencies.cmake]: why commented out?

File: cmake/Dependencies.cmake [hunk 5]
```diff
@@ -171,9 +184,19 @@ list(APPEND DALI_LIBS ${avformat_LIBRARIES})
...
+#      PATH_SUFFIXES lib lib64 lib32
+#      NO_DEFAULT_PATH)
+
+# todo better way to find stubs
+find_library(cuda_Lib cuda
+      PATHS /usr/local/cuda/lib64/stubs
+      NO_DEFAULT_PATH)
+
+# we link cuda with a stub, but dynload nvcuvid (no stub provided)
+# list(APPEND DALI_LIBS ${FFmpeg_LIBS} ${cuda_Lib})
```
Comment (JanuszL) [inline cmake/Dependencies.cmake]: remove

File: dali/pipeline/operators/reader/nvdecoder/nvdecoder.h [hunk 6]
```diff
@@ -15,7 +15,7 @@
 #ifndef DALI_PIPELINE_OPERATORS_READER_NVDECODER_NVDECODER_H_
 #define DALI_PIPELINE_OPERATORS_READER_NVDECODER_NVDECODER_H_
 
-#include <cuda.h>
+// #include <cuda.h>
```
Comment (JanuszL) [inline dali/pipeline/operators/reader/nvdecoder/nvdecoder.h]: ?

File: cmake/Dependencies.cmake (hunk 4 above)
Comment (Kh4L) [reply cmake/Dependencies.cmake]: I forgot to add the `include` for it

Comment (Kh4L) [reply cmake/Dependencies.cmake]: Done

**Comment by Kh4L** [comment]
Build 594822

**Pull request merged**